src/storageService/index.js

```diff
@@ -19,9 +19,7 @@ export const uploadImageAndReturnReference = async (imageURI, onProgress) => {
     const blob = await BlobService.buildBlobFromURI(imageURI)
     const uuid = uuidv4()
     const imageRef = firebase.storage().ref('images').child(uuid).put(blob, metadata)
-    imageRef.on('state_changed', onProgress)
-    await imageRef
-    return uuid
+    return new Promise((resolve, reject) => imageRef.on('state_changed', onProgress, reject, () => resolve(uuid)))
```

Comment:
:thinking: this line is a bit too long IMO. What do you think writing it like this?
```javascript
return new Promise((resolve, reject) => {
  imageRef.on(
    'state_changed',
    onProgress,
    reject,
    () => resolve(uuid)
  )
})
```